src/systems/shader_param/ShaderParam.cc

```diff
@@ -65,11 +66,21 @@ class ignition::gazebo::systems::ShaderParamPrivate
     public: std::vector<std::string> args;
   };
 
-  /// \brief Path to vertex shader
-  public: std::string vertexShaderUri;
+  /// \brief Data structure for storing shader files uri
+  public: class ShaderUri
+  {
+    /// \brief Shader language: glsl or metal
+    public: std::string language;
+
+    /// \brief Path to vertex shader
+    public: std::string vertexShaderUri;
```

Comment:
Do you want to use the URI class from ignition common?

Reply:
Looks like this might be a a local filename, if so maybe use `std::filename::path`?